### PR TITLE
[SYCLomatic] Clarify need of CUDA headers and add links to use with non-Intel devices

### PR DESCRIPTION
**Pull request by daiyaan-ahmed6**
Adds note section in prerequisite explaining the need for CUDA headers to migrate CUDA code to SYCL.

Adds links that provides guidance on using migrated SYCL code with non-Intel devices.

Signed-off-by: Daiyaan Ahmed <daiyaan.ahmed@intel.com>

### CODE REVIEW

**Review by tomflinda — APPROVED** [review]
LGTM

File: README.md [hunk 1]
```diff
@@ -179,6 +181,11 @@ Follow instructions from the link below to build and run tests:
 * SYCL\* 2020 specification:
 [https://www.khronos.org/registry/SYCL/](https://www.khronos.org/registry/SYCL/)
 * More information on oneAPI and DPC++ is available at [https://www.oneapi.com/](https://www.oneapi.com/)
+* Check the following documentation for guidance on using the migrated SYCL code with non-Intel devices using the DPC++ Compatibility Tool:
+
+  * [Targeting nVIDIA GPUs](https://developer.codeplay.com/products/oneapi/nvidia/2023.2.1/guides/get-started-guide-nvidia#use-dpc-to-target-nvidia-gpus)
```
Comment (zhimingwang36) [inline README.md]: use this URL instead, https://developer.codeplay.com/products/oneapi/nvidia/home/.  As the one used need update time to time as it contains the version: 2023.2.1.

Comment (zhimingwang36) [inline README.md]: nVIDIA  ==>  NVIDIA*
AMD  ==> AMD*

File: README.md [hunk 2]
```diff
@@ -179,6 +181,11 @@ Follow instructions from the link below to build and run tests:
...
+
+  * [Targeting AMD GPUs](https://developer.codeplay.com/products/oneapi/amd/2023.2.1/guides/get-started-guide-amd#use-dpc-to-target-amd-gpus)
```
Comment (zhimingwang36) [inline README.md]: https://developer.codeplay.com/products/oneapi/amd/home/  use this URL instead.

File: README.md (hunk 1 above)
Comment (daiyaan-ahmed6) [reply README.md]: Changed to this URL, Done.

File: README.md (hunk 2 above)
Comment (daiyaan-ahmed6) [reply README.md]: Changed to this URL. Done.

File: README.md (hunk 1 above)
Comment (daiyaan-ahmed6) [reply README.md]: Modified to NVIDIA* and AMD*. Done.

**Review by TejaX-Alaghari — APPROVED** [review]

**Review by zhimingwang36 — APPROVED** [review]
LGTM

**Pull request merged**